SPEC.md

```diff
@@ -77,33 +78,58 @@ Network configuration in JSON format is streamed to the plugin through stdin. Th
 
 ### Result
 
-Success is indicated by a return code of zero and the following JSON printed to stdout in the case of the ADD command. This should be the same output as was returned by the IPAM plugin (see [IP Allocation](#ip-allocation) for details).
+Success is indicated by a return code of zero and the following JSON printed to stdout in the case of the ADD command. The `ip` and `dns` items should be the same output as was returned by the IPAM plugin (see [IP Allocation](#ip-allocation) for details) except that the plugin should fill in the `interface` indexes appropriately, which are missing from IPAM plugin output since IPAM plugins should be unaware of interfaces.
 
 ```
 {
   "cniVersion": "0.1.0",
-  "ip4": {
-    "ip": <ipv4-and-subnet-in-CIDR>,
-    "gateway": <ipv4-of-the-gateway>,  (optional)
-    "routes": <list-of-ipv4-routes>    (optional)
-  },
-  "ip6": {
-    "ip": <ipv6-and-subnet-in-CIDR>,
-    "gateway": <ipv6-of-the-gateway>,  (optional)
-    "routes": <list-of-ipv6-routes>    (optional)
-  },
+  "interfaces": [
```

Comment:
I'm unclear how this is going to be used by the container engine (the thing invoking CNI). The interface that is inside the container (one with `sandbox=<netns>`) is quite well-defined. But what about the rest of them? Sounds like another one would be the "host end". But there's no way to tell what this "host end" is for, its role or how it's connected to other components (e.g. `linux-bridge` or `ovs`). Or what happens in case of macvlan, ipvlan, ovs internal port, etc -- these things don't really have a host end.

So if I'm writing a container engine and invoking CNI plugin, what can I do with this information, provided that I don't know if it'll be there or not? Or if it is there, what can be done with that end. I guess it sounds like this is special casing `veth` interfaces and thus leaks the abstraction.